File: c++/meeting-rooms-ii.cc
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());

        int reqRooms = 0;

        priority_queue<int, vector<int>, greater<int>> minHeap;
        
        for (const auto interval: intervals) {
            cout << interval[0] << " " << interval[1] << endl;
            while (!minHeap.empty() && minHeap.top() <= interval[0]) {
                minHeap.pop();
            }

            minHeap.push(interval[1]);
            reqRooms = max(reqRooms, (int) minHeap.size());
        }

        return reqRooms;
    }
};

File: python/meeting-rooms-ii.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        if len(intervals) < 2:
            return len(intervals)
        intervals.sort(key=lambda x: x[0])
        startTimes = [intervals[0][1]]
        for index in range(1, len(intervals)):
            if startTimes[0] <= intervals[index][0]:
                heappop(startTimes)
            heappush(startTimes, intervals[index][1])

        return len(startTimes)
